### ISSUE TITLE
Allow suggested list for Run>Task... to be modified

### ISSUE THREAD

**Issue by CurtisDS** [microsoft/vscode#80649]
First off this is similar to another request https://github.com/microsoft/vscode/issues/30419 but I think different enough it deserves its own post.

I have so many tasks, most of which are just variations of another....

I find the recently used suggestions to mostly hinder my experience as I have to keep searching the list for the task I would like to run.

I would like to see the ability to set the number of "recently used tasks". This way I can limit it to only show 2 or 3.

The other configuration I would like is to be able to always show the tasks in alphabetical order.

Currently, the suggested task list is formatted as such:

```
---------------------------------------------
recently used tasks (sorted by last used)
---------------------------------------------
remaining tasks (sorted alphabetically)
---------------------------------------------
detected tasks
---------------------------------------------
```

Example:

```
---------------------------------------------
build-x-debug             recently used tasks
serve-y-debug
build-y-debug
update-major
build-y-live
serve-x-debug
build-all-prod
serve-x-live
build-x-live
update-minor
build-y-prod
build-y-prod
---------------------------------------------
build-all-test                configure tasks
build-x-prod
serve-y-live
update-patch
---------------------------------------------
npm: install                   detected tasks
npm: e2e
---------------------------------------------
```

In other words if a task is the recently used list it will not be in the centre list, ideally I think it would be beneficial to have this centre list always contain every task that is defined that way you can skim the recently used list but if you miss it there, you will know that it is always going to be in the same spot in that centre list. Along with limiting the number of recently used tasks I believe this will make the suggested list much more usable.

Example:

```
---------------------------------------------
build-x-debug             recently used tasks
serve-y-debug
build-y-debug
---------------------------------------------
build-all-prod                configure tasks
build-all-test
build-x-debug
build-x-live
build-x-prod
build-y-debug
build-y-live
build-y-prod
serve-x-debug
serve-x-live
serve-y-debug
serve-y-live
update-major
update-minor
update-patch
---------------------------------------------
npm: install                   detected tasks
npm: e2e
---------------------------------------------
```

**Comment by alexr00** [comment]
Setting the number of recently used tasks is a good first issue.

**Comment by CurtisDS** [comment]
@hypercubestart unless i missed something. That PR looked to just add the option for turning it on an off. Again, from what I could tell it did not allow you to configure the number of tasks to show as recently used. The issues are related though, if a feature was added to allow you to configure the number of tasks in the recently used list, you could set your preference to 0 and it would also solve the #30419 request to allow you to turn it off entirely.

The second request I raised isnt asking for alphabetical sorting.

What I was trying to get at in my second issue was a configuration to allow tasks to be in the middle section and the recently used section at the same time. Currently if a task is used it gets removed from the middle section and put in the recently used section, instead of remaining in the middle section.

**Comment by hypercubestart** [comment]
@CurtisDS I may be wrong, but I believe that in the PR, the feature is implemented using a number, so it would address your the first request.

My mistake on your second request. I will shortly a submit a PR that will allow tasks to in both sections at the same time. 
I am not sure about the first request, since it seems that #61994 is not being pushed through, so I think it is up to @alexr00 whether this issue addresses it or not

**Comment by alexr00** [comment]
@hypercubestart you are right that #61994 would solve the number of recent tasks by setting it to 0.

@hypercubestart  and @CurtisDS I don't think I want to have a setting for having tasks appear in multiple sections. That isn't something we have in the UI anywhere else and I want to keep tasks consistent with the rest of the UI. An example of where this _doesn't_ happen is the Preference: Color Theme command. If you can find me some examples of where we list things multiple times in quick pick then I'll reconsider :)

Given all that, this is a duplicate of https://github.com/microsoft/vscode/issues/30419

**Comment by alexr00** [comment]
For #61994 there are changes that need to be made before it can be merged. If anyone wants to make those changes then I would look at merging it again.

**Comment by alexr00** [comment]
@hypercubestart These instructions for checking out a PR worked nicely for me: https://coderwall.com/p/z5rkga/github-checkout-a-pull-request-as-a-branch
From there you can make your changes then submit a new PR.

**Comment by MSSR-code** [comment]
Hi @alexr00 is this issue still open to be taken up as a good first issue.

**Comment by alexr00** [comment]
I think @hypercubestart might be taking a look at it. There's already a PR for it, but it's stale an needs some changes.

**Comment by hypercubestart** [comment]
@MSSR-code  I will be submitting a PR shortly

**Comment by alexr00** [comment]
To verify, try setting `"task.quickOpen.history"` then use the Run Task command several times in a folder/workspace. Verify that the value you used in the setting is respected in the task quick pick "recents". Verify that setting the value to 0 results in no "recents" section in the tasks quick pick.
